post/passwordless-authentication-in-directus-flows-with-plivo/index.md

```diff
@@ -0,0 +1,299 @@
+{
+    "otp": "{{$trigger.query.otp}}"
+}
+```
+
+- Click "Done"
+
+## Generating a Directus session token
+
+- We need to add a new node to generate a directus session token.
+- Click the little plus sign on the "Verify OTP Code" node.
+- Set the name to "Generate Session Token" and then click "Run Script".
+- In the script editor, add the following code:
+
+```javascript
+module.exports = async function(data) {
+    
+    var message = data.$last.message;
+    
+    var rand = function() {
+        return Math.random().toString(36).substr(2);
+    };
+
+    var token = function() {
+        return rand() + rand();
+    };
+    
+    if (message == "session validated successfully.") {
+        return {
+            message,
+            token: token()
+        };
+    } else {
+        return {
+            message
+        }
+    }
+}
+```
+
+## Updating the User Account
+
+- We need to add a new node to update the user account with the session token.
+- Click the little plus sign on the "Generate Session Token" node.
+- Set the name to "Update User Account" and then click "Update Data".
+- In the "Collection" field, click the word "Collection" and select "Edit Raw Value" and set the value to "directus_users".
+- Set the Permissions to "Full Access"
+- In the `Payload` section, add the following JSON:
+
+```json
+{
+    "token": "{{$last.token}}"
+}
+```
+
+- In the `Query` section, add the following JSON:
+
+```json
+{
+    "filter": {
+        "otp_session_uuid": {
+            "_eq": "{{$trigger.query.session_uuid}}"
+        }
+    }
+}
+```
+
+- Click `Done`
+
+## Delivering the Directus Session Token to the Client
+
+- We need to add a new node to deliver the Directus session token to the client.
+- Click the little plus sign to the right of the "Update User Account" node.
+- Set the name to "Deliver Session Token" and then click "Run Script".
+- In the script editor, add the following code:
+
+```javascript
+module.exports = async function(data) {
+	return data.generate_session_token;
+}
+```
+
+## Testing the Flow
+
+- Back in the browser and navigate to the Trigger URL for the `Verify OTP Code` flow we noted earlier.
+- Staying in the browser, edit the Trigger URL appending `?otp={YOUR_OTP_CODE}&session_uuid={YOUR_OTP_SESSION}`. Replace {YOUR_OTP_CODE} with the OTP code you received via SMS and {YOUR_OTP_SESSION} with the `otp_session_uuid` you noted earlier.
+- Hit enter and you should see a response from our Directus flow with the `token`. This token is the session token that we will use to authenticate with Directus.
+
+![](verify_flow.png)
+
+## Summary
+
+You have now successfully set up Passwordless Authentication in Directus Flows using the Plivo Verify API. This will allow users to sign in to your Directus project by sending a one-time password (OTP) to their phone number.
+
+Beyond Passwordless Mobile authentication, email and magic link could also work the same way. The only difference is the way the user receives the code.
```

Comment:
```suggestion
You have now successfully set up passwordless authentication in Directus using the Plivo Verify API. This will allow users to sign in to your Directus project by sending a one time password (OTP) to their phone number.

This same general workflow could be used for emails and magic links could also work the same way. It's important to note that static tokens do not expire and are stored in plaintext, so you should consider a strategy for invalidating tokens.
```